presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -769,4 +787,273 @@ private static BigDecimal castTokenAsJavaDecimal(JsonParser parser, int precisio
         }
         return result;
     }
+
+    // given a JSON parser, write to the BlockBuilder
+    public interface BlockBuilderAppender
```

Comment:
JsonUtil is supposed to be a set of `Util`s.

I would separate this class into 3 parts (reader/writer/util) after this pull request is merged.